multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/eventstore/TenantEventSchedulerSegmentFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.axonframework.extensions.multitenancy.components.eventstore;
+
+import org.axonframework.eventhandling.scheduling.EventScheduler;
+import org.axonframework.extensions.multitenancy.components.TenantDescriptor;
+
+import java.util.function.Function;
+
+/**
+ * Factory for creating {@link EventScheduler} instances for a given {@link TenantDescriptor}.
+ *
+ * @author Stefan Dragisic
+ * @since 4.9.0
+ */
+public interface TenantEventSchedulerSegmentFactory extends Function<TenantDescriptor, EventScheduler>{}
```

Comment:
Question: why is "Segment" added to this class? All other factory interfaces are called `Tenant{infrastructure-component-name}Factory`.

Reply:
You're right...I was literally only checking the DLQ implementation, which doesn't have `Segment` in the title. My bad!